### PR TITLE
Add new validation for date of birth

### PR DESCRIPTION
**Pull request by Zain-Abbas1**
- Alter method to run for DoB validation
- New validation for DoB to check it matches format submitted from the web app
- Add new unit tests

### CODE REVIEW

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java [hunk 1]
```diff
@@ -49,4 +53,23 @@ public boolean validateEquals(Object expected, Object actual, String location, E
         }
         return true;
     }
+    
+    public boolean validateDateOfBirth(String dob, String location, Errors errs) {
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java:57]: This should probably be in the PscDiscrepancyValidator class as its field specific not generic.

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java [hunk 2]
```diff
@@ -49,4 +53,23 @@ public boolean validateEquals(Object expected, Object actual, String location, E
...
+        if (validateNotNull(dob, location, errs)) {
+            if (!validateNotEmpty(dob, location, errs)) {
+                return true;
+            } else {
+                try {
+                    final DateTimeFormatter datePattern = DateTimeFormatter.ofPattern("MM/uuuu");
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java:63]: I'm not convinced but should this be a class constant?

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyValidator.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
      */
     public Errors validateForCreation(PscDiscrepancy pscDiscrepancy, Errors errs) {
         validatePscName(errs, pscDiscrepancy.getPscName());
-        validateNotNull(pscDiscrepancy.getPscDateOfBirth(), PSC_DOB, errs);
+        validateDateOfBirth(pscDiscrepancy.getPscDateOfBirth(), PSC_DOB, errs);
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyValidator.java:30]: Could you update Unit tests for this method.

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java [hunk 4]
```diff
@@ -49,4 +53,23 @@ public boolean validateEquals(Object expected, Object actual, String location, E
         }
         return true;
     }
+
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java:56]: Javadocs please.

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyValidator.java (hunk 3 above)
Comment (Zain-Abbas1) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyValidator.java:30]: Done

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java (hunk 2 above)
Comment (Zain-Abbas1) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java:63]: Could do if you want? DateTimeFormatter is thread-safe

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java (hunk 1 above)
Comment (Zain-Abbas1) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java]: Done

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java [hunk 5]
```diff
@@ -49,4 +49,5 @@ public boolean validateEquals(Object expected, Object actual, String location, E
...
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/Validators.java:52]: random new line

**Review by CH-JonathanBlake — APPROVED** [review]

**Pull request merged**